src/agents/common.jl

```diff
@@ -16,6 +16,7 @@ mutable struct HEMData
     index_h::Dimension # customer types
     index_j::Dimension # green tariff technologies
     index_z::Dimension # zone index
+    index_sector::Dimension # zone index
```

Comment:
We probably also want to have the index_h-to-sector mapping (https://github.com/nrel-hem/HolisticElectricityModelData.jl/blob/main/src/create_run_inputs.jl#L580) here so we can refer to it as needed in the rest of the code.